functions/onNewRequestUpdateMetrics/onNewRequestUpdateMetrics.function.js

```diff
@@ -0,0 +1,23 @@
+const admin = require('firebase-admin')
+const functions = require('firebase-functions')
+const moment = require('moment')
+
+const onNewRequestUpdateMetrics = functions
+  .firestore
+  .document('users/{userUID}/requests/{requestUID}')
+  .onCreate((snapshot, context) => {
+    const currentMoment = moment()
+    const startDate = currentMoment.startOf('week').format('YYYY-MM-DD')
+    const endDate = currentMoment.endOf('week').format('YYYY-MM-DD')
+
+    admin
+      .firestore()
+      .doc(`metric-amount-of-requests/${startDate}_${endDate}`)
+      .set({
+        startDate,
+        endDate,
+        amountOfRequests: admin.firestore.FieldValue.increment(1)
+      }, { merge: true })
```

Comment:
Does this code create a new document initialized with `amountOfRequests: 1` if the referenced document doesn't exist?

Reply:
Yes. `set` creates the document if it doesn't exist and `increment` creates the field with a value of `1` if the field doesn't exists.

https://firebase.google.com/docs/reference/js/firebase.firestore.FieldValue#static-increment